github-workflow: Reduce scope of quick-install test

tgraf opened the pull request:
The current quick-install tests deployment of the quickinstall yaml
without building new images and then runs the connectivity check. THis
is problematic as it means that no change will pass the test where a
helm template change depends on a code change.

The connectivity test is already being run in the "conformance-test"
test where images are being built properly.

There is a small diff in the options bein used in the "conformance-test"
and the default options in the quickinstall YAML but nothing that
wouldn't be covered by the overall CI anyway.

Reduce the scope of the quick-install test to validate that the
quickinstall YAML is not outdated.

errordeveloper reviewed (approved):
LGMT!

fristonio commented on .github/workflows/smoke-test.yaml:
```diff
@@ -37,64 +37,6 @@ jobs:
-      - name: Check ${{ matrix.target.template }}
-        run: |
-          make -C install/kubernetes ${{ matrix.target.name }}
-          git diff --exit-code
```
I think we can still keep this section where we check if the `quick-install/experimental-install` is not outdated?

sayboras replied:
agreed with @fristonio. 

I am also thinking about spin off this check, `preflight-clusterrole` and `lint` to another file (e.g. helm-check.yaml) :thinking:

tgraf committed:
smoke-test: Reduce scope of quick-install test to validation

The current quick-install tests deployment of the quickinstall yaml
without building new images and then runs the connectivity check. THis
is problematic as it means that no change will pass the test where a
helm template change depends on a code change.

The connectivity test is already being run in the "conformance-test"
test where images are being built properly.

There is a small diff in the options bein used in the "conformance-test"
and the default options in the quickinstall YAML but nothing that
wouldn't be covered by the overall CI anyway.

Reduce the scope of the quick-install test to validate that the
quickinstall YAML is not outdated.

Signed-off-by: Thomas Graf <thomas@cilium.io>

fristonio reviewed (approved):
LGTM :rocket:
